src/seedu/StudentHandBook/commands/RandomCommand.java

```diff
@@ -0,0 +1,55 @@
+package seedu.StudentHandBook.commands;
+
+import seedu.StudentHandBook.data.person.ReadOnlyPerson;
+
+import java.util.List;
+import java.util.*;
+
+
+public class RandomCommand extends Command {
+
+    public static final String COMMAND_WORD = "random";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ":\n"
+            + "Display a random person and their details in the address book. \n\t"
+            + "Example: " + COMMAND_WORD;
+
+
+    @Override
+
+    public CommandResult execute() {
+        final List<ReadOnlyPerson> randomPerson = getRandomPerson();
+        if(randomPerson.size() == 0) {
+            return new CommandResult(getMessageForRandomFail());
+        }
+        else {
+            return new CommandResult(getMessageForRandomShown(), randomPerson);
+        }
+    }
+
+
+    private List<ReadOnlyPerson> getRandomPerson() {
+        final List<ReadOnlyPerson> randomEntry = new ArrayList<>();
+        Random random = new Random();
+        List<ReadOnlyPerson> allPerson = addressBook.getAllPersons().immutableListView();
+        int upperBound = allPerson.size();
+        if (upperBound == 0) {
+            return randomEntry;
+        }
+
+        int index = 0;
+        int randomIndex = random.nextInt(upperBound);
+
+        for(ReadOnlyPerson person : allPerson) {
```

Comment:
Since you are interested in only one random person, you can directly return allPerson[randomIndex] rather than iterating through the entire list and again adding a single person to a list called randomEntry